lib/autoBlock.js

```diff
@@ -292,6 +399,21 @@ function validateBlockSettings (step, stepName) {
   }
 }
 
+function validateHandlerSettings (step) {
+  var keys = _.keys(step)
+
+  var extra = _.reject(keys, (key) => {
+    return _.includes(validHandlerKeys, key)
+  })
+
+  if (extra.length) {
+    extra = _.map(extra, (key) => {
+      return key
+    })
+    throw new Error('Invalid handler settings: ' + extra.join(', '))
```

Comment:
This function seems to throw if there are extra fields, but it doesn't seem to validate missing fields, such as `func` is that right? Or is `func` optional?

You could possibly do validation like this with joi in a cleaner way.